app/controllers/api/backyards_controller.rb

```diff
@@ -11,8 +11,12 @@ def index
   end
 
   def show
-    backyard_article = Article.find(params[:id])
-    render json: backyard_article, serializer: BackyardsShowSerializerSerializer, root: :backyard_article
+    backyard_article = Article.where(id: params[:id], backyard: true).first
+    if backyard_article
+      render json: backyard_article, serializer: BackyardsShowSerializerSerializer, root: :backyard_article
+    else
+      render json: { error_message: "Couldn\'t find Article with 'id'=#{params[:id]}" }, status: 404
```

Comment:
I'm not sure about this message. Does not seem as good UX

Reply:
We have a rescue in `application controller`

```

rescue_from ActiveRecord::RecordNotFound do |exception|    
  render json: {error_message: exception}, status: 404 
end

```

Reply:
changed to `Backyard Article with 'id'=#{article.id} does not exist`. It is for the guys with postman, you cannot receive this error using the app

Reply:
Actually we don't have a front end rescue from 404

Reply:
![image](https://user-images.githubusercontent.com/75306727/120234937-475bab80-c259-11eb-808f-ad2b35eb1810.png)

Reply:
xD

Reply:
but we use same error handler everywhere, so it is easy to add. Also strange as I thought when there is no case for specific error it passes whatever error API sent